.github/workflows/releases.yml

```diff
@@ -0,0 +1,66 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package to PyPI
+
+on:
+  push:
+    branches: [stable]
+  release:
+    types: [published]
+
+jobs:
+  lint-and-unit-tests:
+    name: Lint & Unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install tox
+    - name: Run lint checkers
+      run: |
+        tox -e lint
+    - name: Run unit tests
+      run: |
+        tox -e unit
```

Comment:
We could also run `tox -e docs` to verify that the documentation builds properly

Reply:
I think there will be another PR for this, where docs will also be published.

Reply:
iirc the actual building/publishing of the docs is done by readthedocs. However running the build job also here would ensure that there's nothing broken with it.

Reply:
This is also a great idea for any PR against master. The expectation is that stable releases will be created from the master branch, so it shouldn't come as a surprise if the docs don't build (except if it's related to a library update or anything different from our changes).